src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -122,4 +130,97 @@ public Stats deserialize(JsonElement jsonElement, Type type,
             }
         }
     }
+
+    /**
+     * Deserializer for Arguments class.
+     * Breaks the incoming arguments into args and kwargs parts
+     * and fills a new Arguments instance.
+     */
+    private class ArgumentsDeserializer implements JsonDeserializer<Arguments> {
+
+        private static final String KWARG_KEY = "__kwarg__";
+
+        @Override
+        public Arguments deserialize(JsonElement json, Type typeOfT,
+                JsonDeserializationContext context) throws JsonParseException {
+            Arguments result = new Arguments();
+
+            if (json != null && json.isJsonArray()) {
+                for (JsonElement jsonElement : json.getAsJsonArray()) {
+                    fillArgs(result, jsonElement);
+                }
+            }
+
+            return result;
+        }
+
+        /**
+         * Fills args/kwargs to given Arguments instance based on JSON data
+         * from jsonElement.
+         *
+         * @param result Arguments to be filled
+         * @param jsonElement input JSON data
+         */
+        private void fillArgs(Arguments result, JsonElement jsonElement) {
+            boolean isKwarg = isKwarg(jsonElement);
+            // we don't need __kwarg__ attribute:
+            // args/kwargs are stored to separate attributes of Arguments class
+            // so we don't need this information anymore
```

Comment:
This is only partially true. Removing `__kwarg__` from objects that are actual kwarg objects is ok since the information is preserved by putting them in a different place in the `Arguments` class. But for normal objects that contain a `__kwarg__` with a value other than `true` (including other types) we lose the information of them being there and what value they had. That said judging from the results of the `test.arg` function the `__kwarg__` with values other than `true` gets dropped from the object and not passed to the function so i guess its ok not to preserve this information.